src/redisai.c

```diff
@@ -121,13 +129,14 @@ int ensureRunQueue(const char* devicestr) {
     run_queue_info->run_queue = queueCreate();
     pthread_cond_init(&run_queue_info->queue_condition_var, NULL);
     pthread_mutex_init(&run_queue_info->run_queue_mutex, NULL);
-    pthread_t tid;
-
-    if (pthread_create(&tid, NULL, RedisAI_Run_ThreadMain, run_queue_info) != 0){
-      freeRunQueueInfo(run_queue_info);
-      return REDISMODULE_ERR;
+    run_queue_info->threads = (pthread_t *)malloc(sizeof(pthread_t) * perqueueThreadPoolSize);
+    /* create threads */
+    for (int i = 0; i != perqueueThreadPoolSize; i++){
```

Comment:
Are you using `!=` (vs `<`) for performance reasons?

Reply:
No @lantiga. Was a typo. Corrected it in the new commit.